File: SDKs/github.com/ghes-3.0/1.1.4/typescript/src/sdk/models/operations/pullscheckifmerged.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";



export class PullsCheckIfMergedPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=owner" })
  owner: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=pull_number" })
  pullNumber: number;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=repo" })
  repo: string;
}


export class PullsCheckIfMergedRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: PullsCheckIfMergedPathParams;
}


export class PullsCheckIfMergedResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/github.com/ghes-2.22/1.1.4/python/src/sdk/models/operations/pulls_check_if_merged.py
from dataclasses import dataclass, field



@dataclass
class PullsCheckIfMergedPathParams:
    owner: str = field(metadata={'path_param': { 'field_name': 'owner', 'style': 'simple', 'explode': False }})
    pull_number: int = field(metadata={'path_param': { 'field_name': 'pull_number', 'style': 'simple', 'explode': False }})
    repo: str = field(metadata={'path_param': { 'field_name': 'repo', 'style': 'simple', 'explode': False }})
    

@dataclass
class PullsCheckIfMergedRequest:
    path_params: PullsCheckIfMergedPathParams = field()
    

@dataclass
class PullsCheckIfMergedResponse:
    content_type: str = field()
    status_code: int = field()
    
